gdx/src/com/badlogic/gdx/utils/Array.java

```diff
@@ -192,6 +192,40 @@ public void swap (int first, int second) {
 		items[second] = firstValue;
 	}
 
+	/** Returns true if the specified value was replaced successfully with the replacement
+	 * @param value May be null.
+	 * @param identity If true, == comparison will be used. If false, .equals() comparison will be used.
+	 * @param replacement the first value will be replaced by this replacement if found
+	 * @return if value was found and replaced */
+	public boolean replaceFirst (@Null T value, boolean identity, T replacement) {
+		T[] items = this.items;
+		if (identity || value == null) {
+			for (int i = 0, n = size; i < n; i++)
+				if (items[i] == value) {
+					items[i] = replacement;
+					return true;
+				}
+		} else {
+			for (int i = 0, n = size; i < n; i++)
+				if (value.equals(items[i])) {
+					items[i] = replacement;
+					return true;
+				}
+		}
+		return false;
+	}
+
+	public void replaceAll (@Null T value, boolean identity, @Null T replacement) {
```

Comment:
I think there's a bit of a mismatch here in how replaceAll() returns void but replaceFirst() returns a usable boolean. Could replaceAll() return false if nothing was changed, or true if anything was replaced? Alternately, true if any value was found, even if the replacement didn't change it.